File: ts-server/src/entity/Member.ts
import { Entity, PrimaryColumn, Column, OneToOne, JoinColumn, ManyToOne, ManyToMany, JoinTable, OneToMany } from 'typeorm';
import { Organization } from './Organization';
import { Category } from './Category';
import { Board } from './Board';
import { Comment } from './Comment';

@Entity()
export class Member {
    @PrimaryColumn({ type: 'char', length: 50 })
    id!: string;

    @ManyToOne(() => Organization, organization => organization.member)
    organization!: Organization;

    @ManyToMany(() => Category, category => category.member)
    @JoinTable()
    category!: Category[];

    @OneToMany(() => Board, board => board.writer)
    board!: Board[];

    @OneToMany(() => Comment, comment => comment.writer)
    comment!: Comment[];

    @Column({ type: 'varchar', length: 50 })
    name!: string;

    @Column({ type: 'varchar', length: 500 })
    password!: string;

    @Column({ type: 'varchar', length: 300, nullable: true })
    image!: string;
}

File: moocServer/src/main/java/com/mooc/moocServer/domain/Member.java
package com.mooc.moocServer.domain;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter(AccessLevel.PRIVATE)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class Member {
    @Id
    @Column(name = "member_id")
    private String id;

    @Column(name = "member_password")
    private String password;

    @Column(name = "member_image")
    private String image;

    @Enumerated(EnumType.STRING)
    private MemberRole role;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "organization_id")
    private Organization organization;

    @OneToMany(mappedBy = "writer")
    private List<Board> boards = new ArrayList<>();

    @OneToMany(mappedBy = "writer")
    private List<Comment> comments = new ArrayList<>();

    // == 생성 메서드 == //
    public static Member createMember(String id, String password, Organization organization){
        Member member = new Member();
        member.setId(id);
        member.setPassword(password);
        member.setImage("default image link");
        member.setRole(MemberRole.NONE);
        member.setOrganization(organization);

        return member;
    }
}
